lib/nodejs/lib/thrift/http_connection.js

```diff
@@ -214,6 +214,18 @@ HttpConnection.prototype.write = function(data) {
   var req = (self.https) ?
       https.request(self.nodeOptions, self.responseCallback) :
       http.request(self.nodeOptions, self.responseCallback);
+
+  //support for timeout
+  var timeout = self.nodeOptions.timeout;
+  if(timeout){
+    req.on('socket', function (socket) {
+        socket.setTimeout(timeout);  
+        socket.on('timeout', function() {
+            req.abort();
+        });
+    });
+  }
```

Comment:
Makes sense.

For newer versions of Node, it can become potentially problematic to do redundant timeout handling.
Current Node implementation seems to be ending up with setting socket timeout, so it may not be a problem for now.

Do you have any idea how to check whether the option is supported ?

Another minor point: we can reduce one indent level by using req.setTimeout(timeout, func...) ?